grakn-graph/src/main/java/ai/grakn/graph/internal/concept/OntologyConceptImpl.java

```diff
@@ -337,21 +329,17 @@ private boolean superLoops(){
      * @return A collection of {@link Rule} for which this {@link OntologyConcept} serves as a hypothesis
      */
     @Override
-    public Collection<Rule> getRulesOfHypothesis() {
-        Set<Rule> rules = new HashSet<>();
-        neighbours(Direction.IN, Schema.EdgeLabel.HYPOTHESIS).forEach(concept -> rules.add(concept.asRule()));
-        return Collections.unmodifiableCollection(rules);
+    public Stream<Rule> getRulesOfHypothesis() {
+        return neighbours(Direction.IN, Schema.EdgeLabel.HYPOTHESIS);
     }
 
     /**
      *
      * @return A collection of {@link Rule} for which this {@link OntologyConcept} serves as a conclusion
      */
     @Override
-    public Collection<Rule> getRulesOfConclusion() {
-        Set<Rule> rules = new HashSet<>();
-        neighbours(Direction.IN, Schema.EdgeLabel.CONCLUSION).forEach(concept -> rules.add(concept.asRule()));
-        return Collections.unmodifiableCollection(rules);
+    public Stream<Rule> getRulesOfConclusion() {
```

Comment:
This is correct. Just much less verbose. This does the exact same thing as before (minus the sets). Basically it's shorter because:

1. `neighbours` already uses a generic to determine the return type. So `concept.asRule()` is not needed.
2. This means that "manually" iterating over the stream and adding it to the new set is not needed either. 

If you look above the exact same change happened for `getRulesOfHypothesis ()`